projects/app/src/dictionary/dictionary.ts

```diff
@@ -163,3 +163,74 @@ export const lookupRadicalsByStrokes = async (strokes: number) =>
 export const radicalStrokes = [
   1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17,
 ];
+
+/**
+ * Converts a single pinyin word written with a tone number suffix to use a tone
+ * mark instead (also converts v to ü).
+ */
+export function convertPinyinWithToneNumberToToneMark(pinyin: string): string {
```

Comment:
this piqued my interest as I couldn't believe this doesn't exist already.

Looking at https://github.com/kevb34ns/pinyinify and https://github.com/quizlet/pinyin-converter  but especially https://github.com/mrchenguozheng/pinyin-tone it might be worth joining forces there? Especially the last one looks fairly complete?

Reply:
cheer I'll take a look, this started out as a copilot generated thing but then i got stucked into it and was having too much fun trying to solve it myself. but yeah swapping it out to something off the shelf is more wise 😄